server/protocols/protocols-imap4/src/main/java/org/apache/james/imapserver/netty/ImapChannelUpstreamHandler.java

```diff
@@ -95,22 +96,41 @@ public ImapChannelUpstreamHandler(String hello, ImapProcessor processor, ImapEnc
         this.imapCommandsMetric = imapMetrics.getCommandsMetric();
     }
 
-    private Logger getLogger(Channel channel) {
-        return new Slf4jLoggerAdapter(new ProtocolSessionLogger("" + channel.getId(), new ProtocolLoggerAdapter(logger)));
+    private Logger getLogger(final ChannelHandlerContext ctx) {
+        return new Slf4jLoggerAdapter(
+            new ContextualLogger(
+                getUserSupplier(ctx),
+                "" + ctx.getChannel().getId(),
+                new ProtocolLoggerAdapter(logger)));
+    }
+
+    private Supplier<String> getUserSupplier(final ChannelHandlerContext ctx) {
+        return () -> {
+            Object o = attributes.get(ctx.getChannel());
+            if (o instanceof IMAPSession) {
+                IMAPSession session = (IMAPSession) o;
+                return session.getUser();
+            }
+            return null;

```

Comment:
ContextualLogger is anyway removed in https://github.com/linagora/james-project/pull/928